src/components/text-editor/link-menu/link-menu.tsx

```diff
@@ -0,0 +1,97 @@
+import { Component, Prop, h, Event, EventEmitter } from '@stencil/core';
+import { EditorTextLink } from '../prosemirror-adapter/menu/types';
+
+/**
+ *
+ */
+@Component({
+    tag: 'limel-link-menu',
+    shadow: true,
+    styleUrl: 'link-menu.scss',
+})
+export class LinkMenu {
+    /**
+     * The link
+     */
+    @Prop()
+    public link: EditorTextLink;
+
+    /**
+     * Open state of the dialog
+     */
+    @Prop()
+    public isOpen: boolean = false;
+
+    /**
+     * Emitted when the menu is closed from inside the component.
+     * (*Not* emitted when the consumer sets the `open`-property to `false`.)
+     */
+    @Event()
+    private cancel: EventEmitter<void>;
+
+    /**
+     * Emitted when the menu is saved from inside the component.
+     */
+    @Event()
+    private save: EventEmitter<void>;
+
+    /**
+     * Emitted when the user inputs new values for the link
+     */
+    @Event()
+    private linkChange: EventEmitter<EditorTextLink>;
+
+    public render() {
+        return [
+            <limel-input-field
+                label="Text"
+                value={this.link?.text || ''}
+                onChange={this.handleLinkTitleChange}
+            />,
+            <limel-input-field
+                label="Link"
+                value={this.link?.href || ''}
+                onChange={this.handleLinkValueChange}
+            />,
+            <div class="actions">
+                <limel-button
+                    label="Cancel" // TODO: use translations
```

Comment:
translations are merged now, you can use them in this component

Reply:
Great! I will add a fixup